core/src/main/scala/kafka/server/KafkaHealthcheck.scala

```diff
@@ -70,19 +72,40 @@ class KafkaHealthcheck(private val brokerId: Int,
   }
 
   /**
-   *  When we get a SessionExpired event, we lost all ephemeral nodes and zkclient has reestablished a
-   *  connection for us. We need to re-register this broker in the broker registry.
+   *  When we get a SessionExpired event, it means that we have lost all ephemeral nodes and ZKClient has re-established
+   *  a connection for us. We need to re-register this broker in the broker registry.
    */
-  class SessionExpireListener() extends IZkStateListener {
+  class SessionExpireListener extends IZkStateListener {
+
+    private[server] val stateToSensorMap = {
+      import KeeperState._
+      val stateToNameMap = Map(
+        Disconnected -> "disconnected",
+        SyncConnected -> "sync-connected",
+        AuthFailed -> "auth-failed",
+        ConnectedReadOnly -> "connected-read-only",
+        SaslAuthenticated -> "sasl-authenticated",
+        Expired -> "expired"
+      )
+      stateToNameMap.map { case (state, stateName) =>
+        val sensorName = s"zk-$stateName"
```

Comment:
Not a big deal, but maybe it is better to use zookeeper instead of zk to make it clear that we are talking about zookeeper. We tend to use zk informally in discussions and in the code, but someone looking at jmx metrics might not necessarily recognize it. Again, not a big deal if we keep it as is.